metrics/nist_mt/nist_mt.py

```diff
@@ -0,0 +1,137 @@
+"""
+
+
+_KWARGS_DESCRIPTION = """
+Computes NIST score of translated segments against one or more references.
+Args:
+    predictions: predictions to score. For sentence-level NIST, a string;
+     for corpus-level NIST, a list of setences (str)
+    references:  potentially multiple references for each prediction.  For sentence-level NIST, a
+     list of potential references (str); for corpus-level NIST, a list (corpus) of lists
+     of potential references (str)
+    n: highest n-gram order
+    tokenize_kwargs: arguments passed to the tokenizer (see: https://github.com/nltk/nltk/blob/90fa546ea600194f2799ee51eaf1b729c128711e/nltk/tokenize/nist.py#L139)
+Returns:
+    'nist_mt': nist_mt score
+Examples:
+    >>> nist_mt = evaluate.load("nist_mt")
+    >>> hypothesis = "It is a guide to action which ensures that the military always obeys the commands of the party"
+    >>> reference1 = "It is a guide to action that ensures that the military will forever heed Party commands"
+    >>> reference2 = "It is the guiding principle which guarantees the military forces always being under the command of the Party"
+    >>> reference3 = "It is the practical guide for the army always to heed the directions of the party"
+    >>> nist_mt.compute(predictions=[hypothesis], references=[[reference1, reference2, reference3]])
+    {'nist_mt': 3.3709935957649324}
+"""
+
+
+@evaluate.utils.file_utils.add_start_docstrings(_DESCRIPTION, _KWARGS_DESCRIPTION)
+class Nist_mt(evaluate.Metric):
+    """A wrapper around NLTK's NIST implementation."""
+
+    def _info(self):
+        return evaluate.MetricInfo(
+            module_type="metric",
+            description=_DESCRIPTION,
+            citation=_CITATION,
+            inputs_description=_KWARGS_DESCRIPTION,
+            features=[
+                datasets.Features(
+                    {
+                        "predictions": Value("string", id="prediction"),
+                        "references": Sequence(Value("string", id="reference"), id="references"),
+                    }
+                ),
+                datasets.Features(
+                    {
+                        "predictions": Sequence(Value("string", id="prediction"), id="predictions"),
+                        "references": Sequence(
+                            Sequence(Value("string", id="reference"), id="references"), id="reference_corpus"
+                        ),
+                    }
+                ),
+            ],
+            homepage="https://www.nltk.org/api/nltk.translate.nist_score.html",
+            codebase_urls=["https://github.com/nltk/nltk/blob/develop/nltk/translate/nist_score.py"],
+            reference_urls=["https://en.wikipedia.org/wiki/NIST_(metric)"],
+        )
+
+    def _compute(self, predictions, references, n: int = 5, lowercase=False, western_lang=True):
+        tokenizer = NISTTokenizer()
+        if isinstance(predictions, str) and isinstance(references[0], str):  # sentence nist_mt
```

Comment:
I think predictions are always a list so this should be 
```suggestion
        if isinstance(predictions[0], str) and isinstance(references[0], str):  # sentence nist_mt
```
no?

Reply:
No, I do not think so because for NIST you can have multiple references, so references also has one dimension more than predictions.

I wanted to account for both single sentences and batches of sentences. But I think you are right: _compute always works on batches, right? So I guess I can just remove these if-clauses. And I think I should then also be able to remove the first feature in self.features. Those features only indicate the type of a batch, right? Not single instances?

Reply:
I think the if/else is fine, but it should be `predictions[0]` in both cases whereas for refs it is `references[0]` and ``references[0][0]` for one or more references.

`self.features` show the type of _one element_ of the batch.

Reply:
I've updated the metric, assuming that a given sample is always prediction (str) and reference (Sequence[str]). That should take away some confusion, I believe.